doc/filters.texi

```diff
@@ -13005,6 +13005,95 @@ lut2='if(lt(x,y),0,if(gt(x,y),pow(2,bdx)-1,pow(2,bdx-1))):if(lt(x,y),0,if(gt(x,y
 @end example
 @end itemize
 
+@section lvpdnn
+
+Inference the content in the input image/video by applying the lvpdnn methods based on
+convolutional neural networks. Supported models:
+
+TensorFlow model files(.pb)
+
+The filter accepts the following options:
+
+@table @option
+@item filter_type
+Specify which filter to use. This option accepts the following values:
+
+@table @samp
+@item lvpclassify
+lvpclassify filter. To conduct lvpclassify filter, you need to use a classify model.
+
+@item lvpodetect
+lvpodetect filter. To conduct object detect filter, you need to use a object detect model.
+@end table
+
+Default value is @samp{lvpclassify}.
+
+@item dnn_backend
+Specify which DNN backend to use for model loading and execution. This option accepts
+the following values:
+
+@table @samp
+@item native
+Native implementation of DNN loading and execution.
+
+@item tensorflow
+TensorFlow backend. To enable this backend you
+need to install the TensorFlow for C library (see
+@url{https://www.tensorflow.org/install/install_c}) and configure FFmpeg with
+@code{--enable-libtensorflow}
+@end table
+
+Default value is @samp{tensorflow}.
+
+@item model
+Set path to model file specifying network architecture and its parameters.
+Note that different backends use different file formats. TensorFlow can load files for only its format.
+
+@item input
+Set the input name of the dnn network.
+
+@item output
+Set the output name of the dnn network.
+
+@item sample
+Set the sample rate for inference.
+
+@item threshold
+Set the threshold for inference.
+
+@item log
+Set the log file path for save inference results.
+
+@end table
+
+@itemize
+@item
+Software transcoding:
+@example
+ffmpeg -i input.mp4 -vf lvpdnn=model=tmodel.pb:input=input:output=outshape:sample=1:threshold=0.5:log=outlog.txt out.mp4
+@end example
+
+@item
+Hardware transcoding:
+@example
+ffmpeg -vsync 0 -hwaccel cuvid -c:v h264_cuvid -i input.mp4 -c:a copy -vf lvpdnn=model=tmodel.pb:input=input:output=outshape:sample=1:threshold=0.5:log=outlog.txt -c:v h264_nvenc -b:v 5M out.mp4
+@end example
+
+@item
+Mixed transcoding:
+@example
+ffmpeg -vsync 0 -c:v h264_cuvid -i input.mp4 -vf lvpdnn=model=tmodel.pb:input=input:output=outshape:sample=1:threshold=0.5:log=outlog.txt -c:v h264_nvenc out.mp4
```

Comment:
Does this command perform software decoding and hardware encoding?

Reply:
Yes, This is a mixed command, that is software decoding and hardware encoding.

Reply:
Can we make note of that just so it is clear for the reader? i.e. "Mixed transcoding (software decoding and hardware encoding)"

Reply:
I might be missing something here but wouldn't setting `-c:v h264_cuvid` before the `-i <input>` cause ffmpeg to use cuvid (hardware) for decoding? https://trac.ffmpeg.org/wiki/HWAccelIntro#NVDECCUVID

Reply:
Thanks for letting me know. I am afraid that I didn't pay extra attention to this example. Just referred to the "Mixing CPU and GPU processing" part in the following [link](https://developer.nvidia.com/blog/nvidia-ffmpeg-transcoding-guide/). I think it would be reasonable to modify it as follows.

@item
Mixed transcoding (software decoding and hardware encoding)
@item
ffmpeg -i input.mp4 -vf lvpdnn=model=tmodel.pb:input=input:output=outshape:sample=1:threshold=0.5:log=outlog.txt -c:v h264_nvenc out.mp4